website/docs/r/lb_listener_rule.html.markdown

```diff
@@ -220,10 +240,50 @@ Authentication Request Extra Params Blocks (for `authentication_request_extra_pa
 * `key` - (Required) The key of query parameter
 * `value` - (Required) The value of query parameter
 
+### Condition Blocks
+
+One of more condition blocks can be set per rule. Most condition types can only be specified once per rule except for `http-header` and `query-string` which can be specified multiple times.
+
 Condition Blocks (for `condition`) support the following:
 
-* `field` - (Required) The name of the field. Must be one of `path-pattern` for path based routing or `host-header` for host based routing.
-* `values` - (Required) The path patterns to match. A maximum of 1 can be defined.
+* `field` - (Required) The type of condition. Valid values are `host-header`, `http-header`, `http-request-method`, `path-pattern`, `query-string` and `source-ip`.
+* `host_header` - (Optional) Host headers to match. Host Header block fields documented below. Required if `field` is `host-header` and `values` is empty.
+* `http_header` - (Optional) HTTP headers to match. HTTP Header block fields documented below. Required if `field` is `http-header`.
+* `http_request_method` - (Optional) HTTP request methods to match. HTTP Request Method block fields documented below. Required if `field` is `http-request-method`.
+* `path_pattern` - (Optional) Path patterns to match. Path Pattern block fields documented below. Required if `field` is `path-pattern` and `values` is empty.
+* `query_string` - (Optional) Query strings to match. Query String block fields documented below. Required if `field` is `query-string`.
+* `source_ip` - (Optional) Source IPs to match. Source IP block fields documented below. Required if `field` is `source-ip`.
+* `values` - (Optional, **DEPRECATED**) List of exactly one pattern to match. Only valid when `field` is `host-header` or `path-pattern`, and the `host_header` and `path_pattern` blocks have not been set.
+
+Host Header Blocks (for `host_header`) support the following:
+
+* `values` - (Required) List of host patterns to match. Only one needs to match for the condition to be satisfied.
+
+HTTP Header Blocks (for `http_header`) support the following:
+
+* `http_header_name` - (Required) Name of HTTP header to search.
+* `values` - (Required) List of header value patterns to match. Only one needs to match for the condition to be satisfied.
```

Comment:
I would perhaps enhance each `values` block with the last part of the description of the API docs:
```suggestion
* `values` - (Required) List of header value patterns to match. Only one needs to match for the condition to be satisfied. If the same header appears multiple times in the request, we search them in order until a match is found. If you specify multiple strings, the condition is satisfied if one of the strings matches the value of the HTTP header. To require that all of the strings are a match, create one condition per string.
```

And thus do it for each `values` block. Thoughts?

Reply:
Should we be directly copying the AWS documentation for current behaviour? Balance between documenting enough for how to use the TF resource and what the AWS API will do.

I can expand the documentation if necessary

Reply:
I think people don't tend to read the API doc, unless the TF one is mistaken, so the more we can provide here, the better it is, and the less issues we can potentially get on the repo I _think_. Does that make sense?

Reply:
I have greatly expanded the docs with extra details from the API docs